ultravox/training/train.py

```diff
@@ -78,6 +81,19 @@ def main() -> None:
 
     transformers.set_seed(args.seed)
 
+    local_rank = int(os.environ.get("LOCAL_RANK", 0))
+    is_master = local_rank == 0
+
+    if args.do_train:
+        train(args)
+
+    if args.do_eval and is_master:
+        gc.collect()
+        torch.cuda.empty_cache()
+        evaluate(args)
```

Comment:
The separation was the best way I knew to make sure all GPU memory is released for oaieval to take over.